pallets/moonbeam-foreign-assets/src/impl_fungibles.rs

```diff
@@ -0,0 +1,183 @@
+				DepositConsequence::Success
+			}
+			(AssetStatus::Active, _) => DepositConsequence::Success,
+			(_, _) => DepositConsequence::Blocked,
+		}
+	}
+
+	fn can_withdraw(
+		asset: Self::AssetId,
+		who: &T::AccountId,
+		amount: Self::Balance,
+	) -> WithdrawConsequence<Self::Balance> {
+		if Self::asset_exists(asset) {
+			let balance = Self::balance(asset, who);
+			if balance >= Self::Balance::from(amount) {
+				WithdrawConsequence::Success
+			} else {
+				WithdrawConsequence::BalanceLow
+			}
+		} else {
+			WithdrawConsequence::UnknownAsset
+		}
+	}
+
+	fn asset_exists(asset: Self::AssetId) -> bool {
+		AssetsById::<T>::contains_key(&asset)
+	}
+}
+
+impl<T: Config> Create<T::AccountId> for Pallet<T> {
+	fn create(
+		_id: Self::AssetId,
+		_admin: T::AccountId,
+		_is_sufficient: bool,
+		_min_balance: Self::Balance,
+	) -> sp_runtime::DispatchResult {
+		sp_runtime::DispatchResult::Err(DispatchError::Other(
+			"Not implemented, must create through create_foreign_asset",
+		))
+	}
+}
+
+impl<T: Config> Unbalanced<T::AccountId> for Pallet<T> {
+	fn handle_dust(_dust: frame_support::traits::fungibles::Dust<T::AccountId, Self>) {}
+	fn set_total_issuance(_asset: Self::AssetId, _amount: Self::Balance) {}
+	fn write_balance(
+		asset: Self::AssetId,
+		who: &T::AccountId,
+		amount: Self::Balance,
+	) -> Result<Option<Self::Balance>, DispatchError> {
+		let balance =
+			EvmCaller::<T>::erc20_balance_of(asset, T::AccountIdToH160::convert(who.clone()))
+				.unwrap_or(U256::zero());
+		let contract_address = Pallet::<T>::contract_address_from_asset_id(asset);
+		match (U256::from(amount), balance) {
+			(amount, balance) if amount == balance => {
+				let as_u128 = balance.saturated_into::<u128>();
+				Ok(Some(Self::Balance::from(as_u128)))
+			}
+			(amount, balance) if amount > balance => {
+				EvmCaller::<T>::erc20_mint_into(
+					contract_address,
+					T::AccountIdToH160::convert(who.clone()),
+					U256::from(amount).saturating_sub(balance),
+				)
+				.map_err(|_| DispatchError::Other("Failed to mint into account"))?;
+				let balance = EvmCaller::<T>::erc20_balance_of(
+					asset,
+					T::AccountIdToH160::convert(who.clone()),
+				)
+				.unwrap_or(U256::zero());
+				let as_u128 = balance.saturated_into::<u128>();
+				Ok(Some(Self::Balance::from(as_u128)))
+			} // Add balance
+			(amount, balance) if amount < balance => {
+				EvmCaller::<T>::erc20_burn_from(
+					contract_address,
+					T::AccountIdToH160::convert(who.clone()),
+					U256::from(balance).saturating_sub(amount),
+				)
+				.map_err(|_| DispatchError::Other("Failed to burn from account"))?;
+				let balance = EvmCaller::<T>::erc20_balance_of(
+					asset,
+					T::AccountIdToH160::convert(who.clone()),
+				)
+				.unwrap_or(U256::zero());
+				let as_u128 = balance.saturated_into::<u128>();
+				Ok(Some(Self::Balance::from(as_u128)))
+			}
+			(_, _) => Err(DispatchError::Other("Invalid amount")),
+		}
+	}
+}
```

Comment:
This trait is very dangerous, we should not add it.

Instead, we should implement `type Paymaster: Pay<Beneficiary = Self::Beneficiary, AssetKind = Self::AssetKind>` for `moonbeam-foreign-assets` and possible not need to implement fungibles